<!-- XML -->
<!--Component of edge-ai-ros-inventory-viz - render people tracker demo to an Image topic-->
<launch>

    <arg name="width" default="960"/>
    <arg name="height" default="540"/>
    <arg name="camera_name" default="/imx390"/>
    <arg name="radar_name" default="/ti_mmwave"/>

    <!-- color conversion for input image_raw for visualization -->
    <node pkg="ti_viz_nodes" type="viz_color_conv_yuv2rgb" name="viz_color_conv_yuv2rgb_node_host_trk" output="screen" args="" required="true">
        <param name="width"             value="$(arg width)"/>
        <param name="height"            value="$(arg height)"/>
        <param name="input_yuv_topic"   value="$(arg camera_name)/image_raw"/>
        <param name="output_rgb_topic"  value="$(arg camera_name)/image_rgb"/>
        <param name="yuv_format"        value="YUV420"/>
        <param name="yuv420_luma_only"  value="false"/>
    </node>

    <node pkg="d3_fusion" type="fusion_cnn_radar.py" name="radar_camera_fusion_node_trk" output="screen">
        <param name="camera_name" value="$(arg camera_name)"/>
        <param name="radar_name" value="$(arg radar_name)"/>
        <remap from="/image_raw" to="$(arg camera_name)/image_rgb"/>
        <remap from="$(arg camera_name)/image_fused" to="$(arg camera_name)/image_fused_trk"/>
    </node>
</launch>

